astropy/wcs/wcs.py

```diff
@@ -894,16 +894,19 @@ def _write_det2im(self, hdulist):
         def write_d2i(num, det2im):
             if det2im is None:
                 return
-            f'{dist}{num:d}',
+
             hdulist[0].header[f'{dist}{num:d}'] = (
                 'LOOKUP', 'Detector to image correction type')
             hdulist[0].header[f'{d_kw}{num:d}.EXTVER'] = (
                 num, 'Version number of WCSDVARR extension')
             hdulist[0].header[f'{d_kw}{num:d}.NAXES'] = (
-                len(det2im.data.shape), 'Number of independent variables in d2im function')
+                len(det2im.data.shape), 'Number of independent variables in D2IM function')
+
+            jth = {1: '1st', 2: '2nd', 3: '3rd'}.get(num, f'{num}th')
```

Comment:
If I'm not mistaken this line needs to go in the loop and use `I=1`, not `num`, otherwise the result is like
```
DP1     = 'AXIS.1: 1' / Axis number of the 1st variable in a CPDIS function     
DP1     = 'AXIS.2: 2' / Axis number of the 1st variable in a CPDIS function   
DP2     = 'AXIS.1: 1' / Axis number of the 2nd variable in a CPDIS function     
DP2     = 'AXIS.2: 2' / Axis number of the 2nd variable in a CPDIS function     


```